File: android/src/main/java/com/margelo/nitro/pdfeditor/HybridPdfImage.kt
package com.margelo.nitro.pdfeditor

import com.facebook.proguard.annotations.DoNotStrip
import com.podofo.android.PdfImage as PodofoImage

@DoNotStrip
class HybridPdfImage(internal val native: PodofoImage) : HybridPdfImageSpec() {
  override val width: Double
    get() = native.width.toDouble()

  override val height: Double
    get() = native.height.toDouble()

  override fun dispose() {
    native.close()
    super.dispose()
  }
}

File: cpp/HybridPdfImage.cpp
#include "HybridPdfImage.hpp"

namespace margelo::nitro::pdfeditor {

double HybridPdfImage::getWidth() {
  return static_cast<double>(_image->GetWidth());
}

double HybridPdfImage::getHeight() {
  return static_cast<double>(_image->GetHeight());
}

} // namespace margelo::nitro::pdfeditor
